### PR TITLE
TRAVIS-15 Support new P/Rs and branches

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: lib/travis/build/addons/sonarcloud.rb [hunk 1]
```diff
@@ -114,56 +103,70 @@ def install_build_wrapper
...
         def run
           sh.echo "Preparing SonarQube Scanner parameters", echo: false, ansi: :yellow
+
           if data.pull_request
-            add_scanner_param("sonar.analysis.mode", "preview")
-            add_scanner_param("sonar.github.repository", data.repository[:slug])
-            add_scanner_param("sonar.github.pullRequest", data.pull_request)
-            add_scanner_param("sonar.github.oauth", "$SONAR_GITHUB_TOKEN")
+            if github_token
+              sh.echo "Github token found, running analysis in the deprecated 'preview' mode. Define the token in the project settings in SonarCloud to benefit from the improved P/R analysis", echo: false, ansi: :yellow
```
Comment (janos-ss) [inline lib/travis/build/addons/sonarcloud.rb]: Cosmetic: other scripts in the project use the spelling "GitHub", never "Github".

File: lib/travis/build/addons/sonarcloud.rb [hunk 2]
```diff
@@ -114,56 +103,70 @@ def install_build_wrapper
...
+              add_scanner_param("sonar.analysis.mode", "preview")
+              add_scanner_param("sonar.github.repository", data.repository[:slug])
+              add_scanner_param("sonar.github.pullRequest", data.pull_request)
+              add_scanner_param("sonar.github.oauth", "$SONAR_GITHUB_TOKEN")
+           else
+              add_scanner_param("sonar.pullrequest.id", data.pull_request)
+              add_scanner_param("sonar.pullrequest.target", data.branch)
```
Comment (janos-ss) [inline lib/travis/build/addons/sonarcloud.rb]: Not sure what `data.branch` refers to. But we don't have `sonar.pullrequest.target`. What we have:

- `sonar.pullrequest.base` : the SonarQube branch to merge into (similar to `sonar.branch.target`)
- `sonar.pullrequest.branch` : the name of the short-living SonarQube branch for this PR. Typically the name of the GitHub branch of the PR, such as `feature/dm/foo-bar-baz` (similar to `sonar.branch.name`)

Comment (dbmeneses) [reply lib/travis/build/addons/sonarcloud.rb]: Ok, so the correct property to be set is `sonar.pullrequest.base`. 
Is `sonar.pullrequest.branch` going to be optional, and the CE fetches it from GitHub using the credentials, if they are set?

Comment (janos-ss) [reply lib/travis/build/addons/sonarcloud.rb]: Making `sonar.pullrequest.branch` optional and have the CE fetch it is a nice idea, and we'll try to include in our current sprint. If we run out of time, then we might require it to be passed, and improve it later.

**Pull request merged**